src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -9496,6 +9509,10 @@ interface AutocompleteData {
   txts: string[];
   enums: NSEnums;
   flags(schema: [string, string | number | boolean | string[]][]): { [key: string]: ScriptArg | string[] };
+  hostname: string;
+  filename: string;
+  /** The processes running on the host */
+  processes: ProcessInfo[];
```

Comment:
Yes, you can dodge the RAM cost this way, but you can already do that to get all servers and scripts. Since the server data is mostly static it is even easier than using this to get the process data for free.

My idea with providing this data is to allow suggestions like pids so you can do `run script.js --tracking scriptPid`. This isn't really possible at the moment.
I can remove it if it really bothers you, but to me this would also mean that server and script data can't be provided as well.

Reply:
It being limited to the current host mitigates many of my concerns. I think it's probably OK.